src/inmanta/data/__init__.py

```diff
@@ -594,36 +471,44 @@ def get_order_by_column_api_name(self) -> str:
         """The name of the column that the results should be ordered by"""
         return self.order_by_column
 
-        # External API
-
+    # External API
     def as_filter(
         self,
         offset: int,
         column_value: Optional[PRIMITIVE_SQL_TYPES] = None,
         id_value: Optional[PRIMITIVE_SQL_TYPES] = None,
-        start: Optional[bool] = True,
+        start: bool = True,
     ) -> Tuple[List[str], List[object]]:
-        """Get the column and id values as filters"""
+        """
+        Produce a filter for this order, to select all record before or after the given id
+
+        :param offset: the next free number to use for query parameters
+        :param column_value: the value for the user specified order
+        :param id_value: the value for the built in order order
+        :param start: is this the start filter? if so, retain all values`  > (column_value, id_value)`
+
+        :return: The filter (as a string) and all associated query parameter values
+        """
         relation = ">" if start else "<"
 
         if column_value is None:
             return [], []
```

Comment:
Do I understand correctly that this class doesn't allow the column to be nullable? I'm still not entirely clear on the semantics of `column_value is None`. In the child implementation we seem to now interpret it as filtering on a value that is `null`, while here we seem to still interpret is as the absence of a filter.

Reply:
added comment